arangod/IResearch/IResearchFeature.cpp

```diff
@@ -1142,6 +1142,27 @@ void IResearchFeature::unprepare() {
   _running.store(false);
 }
 
+void IResearchFeature::reportRecoveryProgress(arangodb::IndexId id,
+                                              std::string_view phase,
+                                              size_t current, size_t total) {
+  TRI_ASSERT(total != 0);
+  auto now = std::chrono::system_clock::now();
+
+  if (id != _progressState.lastReportId ||
+      now - _progressState.lastReportTime >= std::chrono::minutes(1)) {
+    // report progress only when link id changes or one minute has passed
+
+    auto progress = static_cast<size_t>(100.0 * current / total);
+    LOG_TOPIC("d1f18", INFO, TOPIC)
+        << "recovering arangosearch link " << id << ", " << phase
```

Comment:
Here we may have link or inverted index, Maybe use more "generic" entity name 
```suggestion
        << "recovering arangosearch index " << id << ", " << phase
```

Reply:
Done